panos_upgrade_assurance/check_firewall.py

```diff
@@ -870,6 +864,185 @@ def check_ssl_cert_requirements(self, rsa: dict = {}, ecdsa: dict = {}) -> Check
         result.status = CheckStatus.SUCCESS
         return result
 
+    def check_non_finished_jobs(self) -> CheckResult:
+        """Check for any job with status different than FIN.
+
+        # Returns
+
+        CheckResult: Object of [`CheckResult`](/panos/docs/panos-upgrade-assurance/api/utils#class-checkresult) class taking \
+            value of:
+
+        * [`CheckStatus.SUCCESS`](/panos/docs/panos-upgrade-assurance/api/utils#class-checkstatus) when all jobs are in FIN state.
+        * [`CheckStatus.FAIL`](/panos/docs/panos-upgrade-assurance/api/utils#class-checkstatus) otherwise, `CheckResult.reason`
+            field contains information about the 1<sup>st</sup> job found with status different than FIN (job ID and the actual
+            status).
+
+        """
+        result = CheckResult()
+
+        all_jobs = self._node.get_jobs()
+
+        for jid, job in all_jobs.items():
+            if job["status"] != "FIN":
+                result.reason = f"At least one job (ID={jid}) is not in finished state (state={job['status']})."
+                return result
+
+        result.status = CheckStatus.SUCCESS
+        return result
+
+    def _calculate_time_distance(self, schedule_type: str, schedule: dict) -> (int, str):
+        """A method that calculates the time distance between two `datetime` objects.
+
+        :::note
+        This method is used only by [`CheckFirewall.check_scheduled_updates()`](/panos/docs/panos-upgrade-assurance/api/check_firewall#checkfirewallcheck_scheduled_updates) method and it expects some information
+        to be already available.
+        :::
+
+        # Parameters
+
+        schedule_type (str): A schedule type returned by PanOS, can be one of: `every-*`, `hourly`, `daily`, `weekly`,
+            `real-time`.
+        schedule (dict): Value of the `recurring` key in the API response, see [`FirewallProxy.get_update_schedules()`](/panos/docs/panos-upgrade-assurance/api/firewall_proxy#firewallproxyget_update_schedules)
+            documentation for details.
+
+        # Raises
+
+        MalformedResponseException: Thrown then the `schedule_type` is not recognizable.
+
+        # Returns
+
+        tuple(int, str): A tuple containing the calculated time difference (in minutes) and human-readable description.
+
+        """
+        time_distance = 0
+        details = "unsupported schedule type"
+
+        if schedule_type == "daily":
+            occurrence = schedule["at"]
+            next_occurrence = datetime.strptime(f"{str(self._mp_now.date())} {occurrence}", "%Y-%m-%d %H:%M")
```

Comment:
Like commented on the function declaration, we can take a reference date as a parameter instead of referring to the `self._mp_now`.

Reply:
already fixed